### DISCUSSION TITLE
Single user mode without login

### DISCUSSION THREAD

**Discussion by TIAcode** [danny-avila/LibreChat#1311] [Feature Requests & Suggestions]
Running LibreChat in protected network just for myself I think some kind of single-user mode would be quite useful? Maybe automatic login to an account when set. I've set my session expiry and refresh token expiry to >month, but I still need to login again after a while.

Also I'd like to ask that I've used ``` to mark code blocks before with ChatGPT, but Librechat does not format that from user as a code block, is there a way to activate that or am I doing something wrong?

**Comment by danny-avila** [comment]
Thanks for your suggestion!

> Running LibreChat in protected network just for myself I think some kind of single-user mode would be quite useful? Maybe automatic login to an account when set. I've set my session expiry and refresh token expiry to >month, but I still need to login again after a while.

Yes at some point we will get to this but it's not a priority at the moment. One thing I can say right now is if you run the /api/ in dev mode, the refresh token expiry is disabled: `npm run backend:dev` or `NODE_ENV=development node api/server/index.js`

> Also I'd like to ask that I've used ``` to mark code blocks before with ChatGPT, but Librechat does not format that from user as a code block, is there a way to activate that or am I doing something wrong?

I'm not sure what you mean by this, ChatGPT doesn't seem to format this neither in the text box or user message?

![Screenshot 2023-12-08 070304](https://github.com/danny-avila/LibreChat/assets/110412045/b3b5bb56-f0f5-41fd-9b8b-cc5823a612e6)
![Screenshot 2023-12-08 070317](https://github.com/danny-avila/LibreChat/assets/110412045/7498f569-4868-4590-a6e0-10359e0ec43f)

**Reply by TIAcode** [reply]
Thanks, I'll try that.

I misspoke, I've actually mostly used some other UIs for GPT and when I gave a code block with ```` it got formatted just like it would in LibreChat reply. You're correct it doesn't happen in the original UI.

**Reply by gaieges** [reply]
Is that still true?  Tried flipping it to `NODE_ENV=development` and was still prompted for a login

**Comment by TIAcode** [comment]
I had trouble with automated log in as the script on the login page expects the fields to be focused and changed before accepting a form submit, so I made a greasemonkey script. Tested to work with qutebrowser (when saved to `~/.config/qutebrowser/greasemonkey`):

```
// ==UserScript==
// @name         AutoLoginLibreChat
// @namespace    http://yourserver:3080/
// @version      1.0
// @description  Auto login for LibreChat
// @author       Santeri Tarvainen
// @match        http://yourserver:3080/login
// @grant        none
// @run-at       document-start
// ==/UserScript==

(function () {
    'use strict';

    console.log('LibreChat autologin script started.');

    function tryLogin() {
        const username = document.querySelector('input[name="email"]');
        const password = document.querySelector('input[name="password"]');
        const loginButton = document.querySelector('button[data-testid="login-button"]');

        if (username && password && loginButton) {
            console.log('Found LibreChat login form elements.');
            username.focus();
            username.value = 'your@address.com'; // Replace these with

            username.dispatchEvent(new Event('input', { bubbles: true }));
            username.dispatchEvent(new Event('change', { bubbles: true }));
            username.blur();
            password.focus();
            password.value = 'yourPassword'; // your credentials.
            password.dispatchEvent(new Event('input', { bubbles: true }));
            password.dispatchEvent(new Event('change', { bubbles: true }));
            password.blur();

            loginButton.click();

            console.log('LibreChat form submitted.');
            return true;
        } else {
            console.log('LibreChat login form elements not found. Retrying...');
            return false;
        }
    }

    // Try immediately
    if (!tryLogin()) {
        // If not successful, start polling
        const intervalId = setInterval(() => {
            if (tryLogin()) {
                clearInterval(intervalId);
            }
        }, 1000); // Adjust polling interval as needed
    }
})();


```

You have to change lines 7, 25 and 31 to match your configuration.

As I'm also using dedicated qutebrowser instance for LibreChat, I also added these keybindings:
```
bind --mode insert <Return> fake-key <Shift+Return>
bind --mode insert <Control+s> click-element css button[data-testid="send-button"]
bind ,i click-element id prompt-textarea
```
So pressing return changes the line instead of submitting, `,i` focuses the input text field and Control+s from insert mode is used to submit the query. Just  thought I'd share as it makes the autologin easy and gives LibreChat a nice native style feeling.

**Comment by iamoeco** [comment]
Any updates on this? Seems like a feature that I'm looking for and that is implemented in ChatGPT.
